// vinxi/http.js
// Minimal shim for vinxi/http — provides cookie helpers used by TanStack Start server functions.
// In the Vinxi runtime these are provided by the actual Vinxi server layer.
// This shim makes the client build pass by avoiding the unresolved import error.
// At runtime, these get replaced by Vinxi's actual implementation.

export function getCookie(name) {
  if (typeof document !== 'undefined') {
    const match = document.cookie.match(new RegExp('(^| )' + name + '=([^;]+)'));
    return match ? decodeURIComponent(match[2]) : undefined;
  }
  // Server-side — Vinxi provides the actual implementation at build time
  return undefined;
}

export function setCookie(name, value, options = {}) {
  if (typeof document !== 'undefined') {
    let cookie = encodeURIComponent(name) + '=' + encodeURIComponent(value);
    if (options.maxAge) cookie += '; max-age=' + options.maxAge;
    if (options.path) cookie += '; path=' + options.path;
    if (options.secure) cookie += '; secure';
    if (options.httpOnly) cookie += '; httponly';
    if (options.sameSite) cookie += '; samesite=' + options.sameSite;
    document.cookie = cookie;
  }
}

export function deleteCookie(name, options = {}) {
  setCookie(name, '', { ...options, maxAge: 0 });
}

export function getHeader(name) {
  return undefined;
}

export function setHeader(name, value) {}

export function getMethod() { return 'GET'; }
export function getURL() { return '/'; }
